src/components/common/TmSessionExtension.vue

```diff
@@ -87,6 +90,31 @@ export default {
         }
       })
     }
+  },
+  apollo: {
+    addressRole: {
+      query: gql`
+        query polkadotSignIn($address: String!) {
```

Comment:
move to store to avoid douplicate code here and in TmSessionSignIn

Reply:
and TmSessionHardware

Reply:
you mean I use apollo in store right

Reply:
this was actually my first idea, but then I thought it is easier like this

Reply:
I mean, why don't do this for everything else right? We have so many repeated code with `balances`, `rewards` and `delegations` queries for example. We could do the same for all of them. Query in the store. Why not? Maybe because they update much regularly?

Please, explain to me

Reply:
yes we need to reduce duplicated code in general. and I am watching out that we don't make the mistakes of the past again.
in this case it doesn't make sense. you are passing the role anyway to the store. so you have 3 places doing the same things but passing it to the same function in the vuex store.